jpo-ode-svcs/src/main/java/us/dot/its/jpo/ode/traveler/TimDepositController.java

```diff
@@ -62,261 +44,259 @@
-    * @param verb
-    * @return
-    */
-   public synchronized ResponseEntity<String> depositTim(String jsonString, RequestVerb verb) {
-
-      if (null == jsonString || jsonString.isEmpty()) {
-         String errMsg = "Empty request.";
-         logger.error(errMsg);
-         return ResponseEntity.status(HttpStatus.BAD_REQUEST).body(JsonUtils.jsonKeyValue(ERRSTR, errMsg));
-      }
-
-      OdeTravelerInputData odeTID = null;
-      ServiceRequest request;
-      try {
-         // Convert JSON to POJO
-         odeTID = (OdeTravelerInputData) JsonUtils.jacksonFromJson(jsonString, OdeTravelerInputData.class, true);
-         if (odeTID == null) {
+    private static final TimIngestTracker INGEST_MONITOR = TimIngestTracker.getInstance();
+
+    private static final String ERRSTR = "error";
+    private static final String WARNING = "warning";
+    private static final String SUCCESS = "success";
+
+    private final Asn1CoderTopics asn1CoderTopics;
+    private final PojoTopics pojoTopics;
+    private final JsonTopics jsonTopics;
+
+    private final SerialId serialIdJ2735;
+    private final SerialId serialIdOde;
+
+    private final MessageProducer<String, String> stringMsgProducer;
+    private final MessageProducer<String, OdeObject> timProducer;
+
+    private final boolean dataSigningEnabledSDW;
+
+    public static class TimDepositControllerException extends Exception {
+
+        private static final long serialVersionUID = 1L;
+
+        public TimDepositControllerException(String errMsg) {
+            super(errMsg);
+        }
+
+    }
+
+    @Autowired
+    public TimDepositController(OdeKafkaProperties odeKafkaProperties, Asn1CoderTopics asn1CoderTopics, PojoTopics pojoTopics, JsonTopics jsonTopics, TimIngestTrackerProperties ingestTrackerProperties) {
+        super();
+
+        this.asn1CoderTopics = asn1CoderTopics;
+        this.pojoTopics = pojoTopics;
+        this.jsonTopics = jsonTopics;
+        this.serialIdJ2735 = new SerialId();
+        this.serialIdOde = new SerialId();
+
+        this.stringMsgProducer = MessageProducer.defaultStringMessageProducer(odeKafkaProperties.getBrokers(),
+                odeKafkaProperties.getProducerType(), odeKafkaProperties.getDisabledTopics());
+        this.timProducer = new MessageProducer<>(odeKafkaProperties.getBrokers(), odeKafkaProperties.getProducerType(),
+                null, OdeTimSerializer.class.getName(), odeKafkaProperties.getDisabledTopics());
+
+        this.dataSigningEnabledSDW = System.getenv("DATA_SIGNING_ENABLED_SDW") == null || System.getenv("DATA_SIGNING_ENABLED_SDW").isEmpty() || Boolean.parseBoolean(System.getenv("DATA_SIGNING_ENABLED_SDW"));
+
+        // start the TIM ingest monitoring service if enabled
+        if (ingestTrackerProperties.isTrackingEnabled()) {
+            log.info("TIM ingest monitoring enabled.");
+
+            ScheduledExecutorService scheduledExecutorService = Executors.newSingleThreadScheduledExecutor();
+
```

Comment:
note: with the introduction of configs, and the [merge conflicts](https://github.com/CDOT-CV/jpo-ode/pull/112/commits/9794fbd3439355f32d645708dc6066d4d7924947) that arose when the ingest monitor PR was merged, I moved the configuration validation shown here into [a validator](https://github.com/CDOT-CV/jpo-ode/pull/112/files/6ef8ce38cc069045e8a36f7d2426be2a135b2752..9794fbd3439355f32d645708dc6066d4d7924947#diff-79bd31eeeb3787cb08e43153c2e7b8669d398e2900efc04bbc1bbef97dc5adf5R5-R20)